### PR TITLE
Issue #2481 use current crc location to determine the msi installation path

### PR DESCRIPTION
**Pull request by anjannath**
Since both are copied to the same location by the msi it is expected to find
crc-tray in the same directory

Fixes: #2481

Relates to: PR https://github.com/code-ready/tray-windows/pull/89

## Testing

Install crc using the msi in location other then `C:\Program Files\CodeReady Containers` and then running `crc setup` should not fail.

(Note: the tray will be started but will immediately fail with an error saying unable to find crc.exe this is fixed by https://github.com/code-ready/tray-windows/pull/89)

### CODE REVIEW

File: pkg/crc/constants/constants.go [hunk 1]
```diff
@@ -200,5 +200,7 @@ func GetCRCWindowsTrayDownloadURL() string {
 }
 
 func GetMsiInstallPath() string {
-	return filepath.Join(os.Getenv("PROGRAMFILES"), "CodeReady Containers")
+	// In case of error path will be empty string and upperr layer will handle
+	currentExecutablePath, _ := os.Executable()
```
Comment (guillaumerose) [inline pkg/crc/constants/constants.go:204]: Maybe log the error ? Otherwise lgtm

Comment (anjannath) [reply pkg/crc/constants/constants.go:204]: should we do logging in the constants package? there's an import cycle if i try to import the logging package :(

Comment (anjannath) [reply pkg/crc/constants/constants.go]: Updated, logging the error now and fixed the import cycle in the next commit.

File: cmd/crc-embedder/cmd/root.go [hunk 2]
```diff
@@ -26,6 +26,10 @@ when building the crc executable for release`,
 }
 
 func init() {
+	err := constants.EnsureBaseDirectoriesExist()
+	if err != nil {
+		fmt.Println("CRC base directories are missing: ", err)
+	}
```
Comment (anjannath) [inline cmd/crc-embedder/cmd/root.go:32]: Should `return` or `os.Exit` maybe?

Comment (guillaumerose) [reply cmd/crc-embedder/cmd/root.go:32]: yes as you wish. It is not production code. crc does `logging.Fatal()`.

File: pkg/crc/logging/logging.go [hunk 3]
```diff
@@ -18,10 +17,6 @@ var (
 )
 
 func OpenLogFile(path string) (*os.File, error) {
-	err := constants.EnsureBaseDirectoriesExist()
```
Comment (guillaumerose) [inline pkg/crc/logging/logging.go:21]: good to remove this!

File: cmd/crc-embedder/cmd/root.go (hunk 2 above)
Comment (anjannath) [reply cmd/crc-embedder/cmd/root.go:33]: added `return`

Comment (guillaumerose) [reply cmd/crc-embedder/cmd/root.go:33]: If you add return here, it won't add the flag and continue the execution ? I think you should either use logging.Fatal or os.Exit but not return.

Comment (anjannath) [reply cmd/crc-embedder/cmd/root.go:34]: updated, used `os.Exit(1)`

**Review by guillaumerose — APPROVED** [review]

**Comment by guillaumerose** [comment]
/retest
retest this please

**Pull request merged**